clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -789,12 +789,43 @@ public boolean matches(AbstractRequest body) {
         assertEquals(new HashSet<>(Arrays.asList(tp1, tp2)), subscriptions.assignedPartitions());
     }
 
+    @Test
+    public void testRebalanceAfterPartitionUnavailable() {
```

Comment:
This one tests the metadata refresh path for unavailable partitions/topics, couldn't think of any other untested paths. I will add an integration test in `core` where the topic is created after the initial poll.